src/map.rs

```diff
@@ -66,35 +66,45 @@ impl<'a, R: 'a + Read> de::MapVisitor for MapVisitor<'a, R> {
         }
     }
 
-    fn size_hint(&self) -> (usize, Option<usize>) {
-        (self.attrs.size_hint().0, None)
+    fn size_hint(&self) -> Option<usize> {
+        self.attrs.size_hint().1
     }
 }
 
 struct AttrValueDeserializer(String);
 
-impl de::Deserializer for AttrValueDeserializer {
+impl<'de> de::Deserializer<'de> for AttrValueDeserializer {
     type Error = Error;
 
-    fn deserialize<V: Visitor>(self, visitor: V) -> VResult<V> {
+    fn deserialize_any<V: Visitor<'de>>(self, visitor: V) -> VResult<V::Value> {
         visitor.visit_string(self.0)
     }
 
-    fn deserialize_enum<V: Visitor>(self, _name: &str, _variants: &'static [&'static str], visitor: V) -> VResult<V> {
+    fn deserialize_u8<V: de::Visitor<'de>>(self, visitor: V) -> VResult<V::Value> {
+        let u = self.0.parse().map_err(Error::ParseIntError)?;
```

Comment:
FWIW migrated tests are failing on master as well - I've added them as TODO for easier migration from serde-xml, but it's not something fixed yet nor we will necessarily support all of them.